### PR TITLE
UD-284-manage-mentors-mentees-matches

### PR DESCRIPTION
**Pull request by JeffDesch**
## Description
Implements the desired DS functions of the UD-284 user stories:
Adds a new "mentee_archive" field in addition to the existing "mentee_ids" field in the Matches documents. Adding a match will add the mentee id to both array fields, while deleting the match will only remove from the "mentee_ids" field leaving an archived record of the old mentee id.
The get_match route was refactored to move id-list retrieval to a separate helper function which can optionally read from the archive field (though as of now nothing is set up to use this ability)
Changes have been tested using the matches integration tests notebook under /tests/

## Jira Ticket
[UD-284](https://bloomtechlabs.atlassian.net/browse/UD-284?atlOrigin=eyJpIjoiYTUyZDk0MGY5NDAzNDcxYmE3NTZhZWI0YTUxZGFlNDYiLCJwIjoiaiJ9)

## Type of Change
- [x] New feature

## Checklist
- [x] My code follows PEP8 style guide
- [x] I have removed unnecessary print statements from my code
- [x] I have made corresponding changes to the documentation if necessary
- [x] My changes generate no errors
- [x] No commented-out code
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

[UD-284]: https://bloomtechlabs.atlassian.net/browse/UD-284?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by morganmallen — APPROVED** [review]

**Review by BrokenShell — APPROVED** [review]
Looking great!

File: app/routers/match_router.py [hunk 1]
```diff
@@ -40,6 +41,25 @@ async def delete_match(data: MatchUpdate):
...
+            ids = Router.db.first("Matches", {"mentor_id": profile_id})["mentee_ids"]
+    elif profile_type == "mentee":
+        if use_archive:
+            ids = [mentor["mentor_id"] for mentor in Router.db.read(
+                "Matches", {"mentee_archive": profile_id})]
+        else:
+            ids = [mentor["mentor_id"] for mentor in Router.db.read(
+                "Matches", {"mentee_ids": profile_id})]
+
+    return ids
```
Comment (richardsgs27) [inline app/routers/match_router.py:60]: I love what you did with this code. Great job!

File: app/routers/match_router.py [hunk 2]
```diff
@@ -21,7 +21,8 @@ async def create_match(data: MatchUpdate):
     return {
         "result": Router.db.upsert_to_set_array("Matches",
                                                 {"mentor_id": data.mentor_id},
-                                                {"mentee_ids": data.mentee_id})
+                                                {"mentee_ids": data.mentee_id,
+                                                 "mentee_archive": data.mentee_id})
```
Comment (richardsgs27) [inline app/routers/match_router.py:25]: Your code is incredible. I can tell how hard you've worked on it. It shows!

File: app/routers/match_router.py [hunk 3]
```diff
@@ -49,17 +69,12 @@ async def get_match(data: MatchQuery):
...
-            "profile_id": {
-                "$in": [mentor["mentor_id"] for mentor in Router.db.read(
-                    "Matches", {"mentee_ids": data.user_id})]
-            }
-        }
     else:
         raise ValueError("get_match: user_type must be 'mentor' or 'mentee'")
 
+    match_ids = get_match_ids(data.user_id, data.user_type)
+    user_query = {"profile_id": {"$in": match_ids}}
```
Comment (richardsgs27) [inline app/routers/match_router.py:78]: I'm definitely taking notes from the way you work. I want my work to be that good. Love it!

**Pull request merged**